src/org/zaproxy/zap/extension/api/CoreAPI.java

```diff
@@ -838,9 +843,21 @@ public void process(Alert alert) {
 			processAlerts(
 					this.getParam(params, PARAM_BASE_URL, (String) null), 
 					this.getParam(params, PARAM_START, -1), 
-					this.getParam(params, PARAM_COUNT, -1), counter);
+					this.getParam(params, PARAM_COUNT, -1), 
+					this.getParam(params, PARAM_RISK, -1), counter);
 			
 			result = new ApiResponseElement(name, Integer.toString(counter.getCount()));
+		} else if (VIEW_ALERTS_SUMMARY.equals(name)) {
+			final ApiResponseList resultList = new ApiResponseList(name);
+			Map<String, Object> alertData = new HashMap<>();
+			int[] riskIdArray = {Alert.RISK_HIGH, Alert.RISK_MEDIUM, Alert.RISK_LOW, Alert.RISK_INFO};
+			CounterProcessor<Alert> counter = new CounterProcessor<>();
+			for (int riskId:riskIdArray) {
+				processAlerts(this.getParam(params, PARAM_BASE_URL, (String) null), -1, -1, riskId, counter);
```

Comment:
The counter needs to be reset before each count, otherwise the following risks include the counts of the previous ones. But, probably better to have a new counter that counts all the risks at the same time, that would allow to just read the alerts once, instead of now that it reads all alerts four times (once for each risk).